tests/src/testApplicationFactory.cc

```diff
@@ -0,0 +1,114 @@
+#define BOOST_TEST_MODULE TestApplicationFactory
+// Only after defining the name include the unit test header.
+#include "ApplicationFactory.h"
+#include "ReferenceTestApplication.h"
+
+#include <boost/test/included/unit_test.hpp>
+
+using namespace boost::unit_test_framework;
+using namespace ChimeraTK;
+
+BOOST_AUTO_TEST_CASE(testNoFactoryNoInstance) {
+  try {
+    (void)ApplicationBase::getInstance();
+    BOOST_ERROR("ApplicationBase::getInstance() did not throw as expected.");
+  }
+  catch(ChimeraTK::logic_error& e) {
+    std::cout << "Exception message debug printout: " << e.what() << std::endl;
+  }
```

Comment:
Why don't you use BOOST_CHECK_THROW for this?

Reply:
I want to see the exception message for a manual check in the test output. We decided that we formally don't check for contents of exception messages, so I do this manually.